rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -39,33 +39,75 @@ pub fn check_content_type<B>(
     }
 }
 
+pub fn accept_header_classifier<B>(req: &RequestParts<B>, content_type: &'static str) -> bool {
+    if req.headers().is_none() {
+        return true;
+    }
+    let headers = req.headers().unwrap();
+    if !headers.contains_key(http::header::ACCEPT) {
+        return true;
+    }
+    let content_type = content_type
+        .parse::<mime::Mime>()
+        .expect("BUG: content_type is not valid");
+    headers
+        .get_all(http::header::ACCEPT)
+        .into_iter()
+        .flat_map(|header| {
+            header
+                .to_str()
+                .ok()
+                .into_iter()
+                .flat_map(|s| s.split(",").map(|typ| typ.split(";").nth(0).unwrap().trim()))
+        })
+        .any(|h| {
+            h.parse::<mime::Mime>()
+                .map(|mim| {
+                    let typ = content_type.type_();
+                    let subtype = content_type.subtype();
+                    match (mim.type_(), mim.subtype()) {
+                        (t, s) if t == typ && s == subtype => true,
+                        (t, mime::STAR) if t == typ => true,
+                        (mime::STAR, mime::STAR) => true,
+                        _ => false,
+                    }
+                })
+                .unwrap_or(false)
+        })
```

Comment:
I think this piece of code deserve some extensive documentation. It's pretty hard to read, there is a lot going on between `map`, `flat_map`, multiple splits and so on.

Is the unwrap after the `nth(0)` safe?

Reply:
> Is the unwrap after the nth(0) safe?

It should return an empty string for an empty string, so safe.